File: examples/sample.swift
import Foundation

protocol Highlighter {
    func highlight(line: String) -> String
}

class SwiftHighlighter: Highlighter {
    private let language: String
    private var inBlockComment = false

    init(language: String) {
        self.language = language
    }

    func highlight(line: String) -> String {
        if inBlockComment {
            if let endRange = line.range(of: "*/") {
                inBlockComment = false
                return String(line[...endRange.upperBound])
            }
            return line
        }

        if line.hasPrefix("//") {
            return line
        }

        if line.contains("/*") {
            if !line.contains("*/") {
                inBlockComment = true
            }
            return line
        }

        return line
    }
}

enum Token {
    case keyword(String)
    case string(String)
    case number(Int)
    case comment(String)
    case plain(String)
}

struct Range {
    let start: Int
    let end: Int

    var length: Int {
        return end - start
    }
}

func createHighlighter(for language: String) -> Highlighter {
    switch language {
    case "swift":
        return SwiftHighlighter(language: language)
    default:
        fatalError("Unknown language: \(language)")
    }
}

let languages = ["swift", "rust", "go", "javascript"]
for lang in languages {
    print("Language: \(lang)")
}

let result: [String] = languages.compactMap { lang in
    guard !lang.isEmpty else { return nil }
    return lang.uppercased()
}

File: examples/sample.rs
use std::collections::HashMap;
use std::fmt;

#[derive(Debug, Clone)]
struct Highlighter {
    language: String,
    in_block_comment: bool,
}

impl Highlighter {
    fn new(language: &str) -> Self {
        Self {
            language: language.to_string(),
            in_block_comment: false,
        }
    }

    fn highlight_line(&mut self, line: &str) -> String {
        if self.in_block_comment {
            if let Some(end) = line.find("*/") {
                self.in_block_comment = false;
                return format!("{}{}",&line[..end + 2], &line[end + 2..]);
            }
            return line.to_string();
        }

        if line.starts_with("//") {
            return line.to_string();
        }

        if let Some(start) = line.find("/*") {
            if line[start + 2..].find("*/").is_none() {
                self.in_block_comment = true;
            }
            return line.to_string();
        }

        line.to_string()
    }
}

impl fmt::Display for Highlighter {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "Highlighter({})", self.language)
    }
}

trait SyntaxEngine {
    fn process(&self, input: &str) -> Vec<String>;
}

struct Token {
    kind: TokenKind,
    value: String,
}

enum TokenKind {
    Keyword,
    String,
    Number,
    Comment,
    Plain,
}

fn main() {
    let mut map: HashMap<&str, i32> = HashMap::new();
    map.insert("rust", 1);
    map.insert("go", 2);

    let keys: Vec<_> = map.keys().collect();
    println!("Languages: {:?}", keys);

    let mut h = Highlighter::new("rust");
    let result = h.highlight_line("fn main() {}");
    println!("{}", result);

    let _ = match result.as_str() {
        "fn main() {}" => true,
        _ => false,
    };
}

/* Block comment
   spans multiple lines
*/
